### PR TITLE
Integración de api de Weets

### PR DESCRIPTION
**Pull request by minotad**
## Summary

Connection to an external geek-joke-api api is made.

## Known Issues

None

## Trello Card

https://trello.com/c/xdL5bamg/32-servicios-promises-witter

### CODE REVIEW

File: WTraining/app/services/weet.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+const axios = require('axios');
```
Comment (marioportela1993) [inline WTraining/app/services/weet.js]: We can to create a request helper to do requests external apis in order to wrap the axios package to be used to any service.

File: WTraining/app/services/weet.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+const logger = require('../logger');
+
+exports.getWeet = () =>
+  axios
+    .get('https://geek-jokes.sameerkumar.website/api?format=json')
```
Comment (marioportela1993) [inline WTraining/app/services/weet.js]: Please set this URL as env variable :)

File: WTraining/app/services/weet.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    .then(res => res.data)
+    .catch(e => logger.error(e));
```
Comment (marioportela1993) [inline WTraining/app/services/weet.js]: Please throw and error to be catched in the controller

**Review by marioportela1993 — CHANGES_REQUESTED** [review]
Nice code! only few comments, please check and let me know if you have questions

File: WTraining/app/services/weet.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+const { requestGet } = require('../helpers/request');
+const logger = require('../logger');
+
+const URL = process.env.ULR_WEET;
```
Comment (marioportela1993) [inline WTraining/app/services/weet.js]: Add this variable (ULR_WEET) in config/index.js file and retrieve the value from there

**Review by marioportela1993 — CHANGES_REQUESTED** [review]

File: WTraining/app/services/weet.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+const { requestGet } = require('../helpers/request');
+const logger = require('../logger');
+const config = require('../../config').common.url;
+
+exports.getWeet = () =>
+  requestGet(config.urlWeet)
+    .then(res => res.data)
+    .catch(e => {
+      logger.error(e);
+      throw e;
```
Comment (emeneses4) [inline WTraining/app/services/weet.js]: You could to throw a specific error using the Wolox error definition. I think that You should created a error in `app/errors.js `and then to define code status error in `app/middleware/errors.js ` for example 

`errors.js `
` exports.VALIDATION_ERROR = 'validation_error';
  exports.validationError = message => internalError(message, exports.VALIDATION_ERROR);
`

`middleware/errors.js
`
`
const statusCodes = {
  [errors.VALIDATION_ERROR]: 422
};
`

**Review by emeneses4 — CHANGES_REQUESTED** [review]

File: WTraining/config/index.js [hunk 6]
```diff
@@ -44,6 +44,9 @@ const config = {
       apiDate: process.env.API_DATE || 'X-API-Date',
       packageVersion: process.env.PACKAGE_VERSION || 'X-Package-Version',
       nodeVersion: process.env.NODE_VERSION || 'X-Node-Version'
+    },
+    url: {
```
Comment (marioportela1993) [inline WTraining/config/index.js]: Please be more specific with the naming of variables. Here maybe is better: 

```
weetService: {
     url: process.env.ULR_WEET
}

**Review by marioportela1993 — CHANGES_REQUESTED** [review]

**Review by marioportela1993 — APPROVED** [review]

**Review by emeneses4 — APPROVED** [review]

**Pull request merged**